### PR TITLE
feat: add InlineMessage component

### PR DESCRIPTION
**Pull request by joshblack**
Context https://github.com/github/primer/issues/3319

#### New

- Add `InlineMessage` component

### Rollout strategy

- [x] Minor release

### Testing & Reviewing

- [ ] Verify component appearance and behavior based on [usage docs](https://github.com/primer/design/pull/746)
- [ ] Evaluate component API, provide feedback if something is misaligned with system expectations

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [x] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge

### CODE REVIEW

**Review by langermank — COMMENTED** [review]
Just a few things! Looking good

File: packages/react/src/InlineMessage/InlineMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+`
+
+const variantToIcon: Record<MessageVariant, React.ReactNode> = {
+  warning: <AlertIcon className="InlineMessageIcon" />,
+  critical: <AlertIcon className="InlineMessageIcon" />,
+  success: <IssueClosedIcon className="InlineMessageIcon" />,
+  unavailable: <AlertIcon className="InlineMessageIcon" />,
+}
+
+const variantToFilledIcon: Record<MessageVariant, React.ReactNode> = {
```
[the remark this answers is not in the document]
Comment (joshblack) [reply packages/react/src/InlineMessage/InlineMessage.tsx:67]: Makes total sense, thanks for clarifying!

File: packages/react/src/InlineMessage/InlineMessage.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+export default meta
+
+export const Default = () => {
+  return <InlineMessage variant="unavailable">An example inline message</InlineMessage>
+}
+
+export const Playground: StoryObj<typeof InlineMessage> = {
+  render(args) {
+    return <InlineMessage {...args}>An example inline message</InlineMessage>
```
[the remark this answers is not in the document]
Comment (joshblack) [reply packages/react/src/InlineMessage/InlineMessage.stories.tsx:18]: Love it, thanks for the suggestion 💯

**Review by lukasoppermann — APPROVED** [review]
Perfect, thank you. 🙏

**Comment by langermank** [comment]
Just a few last things:

- The icon should be the outline variation at the default size, and filled at the small size
- I think it would be nice to align the icon with the first line for improved wrapping (like how you've done it in the Banner component)

Other than that looking fab!

**Review by langermank — APPROVED** [review]
Looks so good! ✨

**Pull request merged**